include/cudawrappers/cu.hpp

```diff
@@ -7,13 +7,25 @@
 #include <iomanip>
 #include <map>
 #include <memory>
+#include <sstream>
 #include <stdexcept>
 #include <string>
 #include <utility>
 #include <vector>
 
+#if !defined(__HIP__)
 #include <cuda.h>
 #include <cuda_runtime_api.h>
+#endif
+
+#if defined(__HIP__) || defined(__HIP_PLATFORM_AMD__)
```

Comment:
I don't think the extra part of the `if` is needed, is it?
```suggestion
#if defined(__HIP__)
```

Reply:
It shouldn't be needed indeed, I added it at some point because my code compiled with g++ and cmake did add HIP_PLATFORM_AMD but __HIP__ wasn't defined. That was an issue on my end though, just _HIP__ should be enough